lib/Net/Async/Blockchain/ETH.pm

```diff
@@ -375,22 +416,40 @@ async sub transform_transaction {
         for my $tx (@transactions) {
             # set the type for each transaction
             # from and to => internal
-            # to => received
-            # from => sent
+            # to => receive
+            # from => send
             my $tx_type_response = await $self->_set_transaction_type($tx);
             $self->source->emit($tx_type_response) if $tx_type_response;
         }
 
-    }
-    catch {
+    } catch {
         my $err = $@;
-        warn sprintf("Error processing transaction: %s, error: %s", $decoded_transaction->{hash}, $err);
+        warn sprintf("Error processing transaction: %s, error: %s", $txn_hash, $err);
         return 0;
     }
 
     return 1;
 }
 
+=head2 _transform_unprocessed_transactions
+
+To send the unprocessed transaction to transform_transaction sub to process.
+
+=cut
+
+async sub _transform_unprocessed_transactions {
+    my ($self) = @_;
+    my ($response, $transaction);
+
+    while (1) {
+        # get eth::subscription::unprocessed_transaction from redis
+        await $self->redis_client->connected;
+        $response    = await $self->redis_client->rpop($redis_key);
+        $transaction = decode_json_utf8($response) if $response;
+        await $self->transform_transaction($transaction, $transaction->{timestamp}) if $transaction;
+    }
```

Comment:
IMO, Better to wait for at least one minute before the next run, because keep connecting to Redis at this high frequency it could be an issue